cafebabel/static/js/articles-tags.js

```diff
@@ -1,70 +1,82 @@
 class Tags {
-  constructor() {
-    this.context = document.querySelector('.tags')
-    this.values = Array.from(this.list.querySelectorAll('input')).map(
-      input => input.value
-    )
-  }
-  get language() {
-    const languages = document.querySelector('#language')
-    return languages.options[languages.selectedIndex].value
-  }
-  get list() {
-    return this.context.querySelector('.tags-list')
+  get context() {
+    return document.querySelector('.tags')
   }
   get removeButtons() {
-    return this.list.querySelectorAll('li button')
+    return this._list.querySelectorAll('li button')
   }
   get fieldAdd() {
     return this.context.querySelector('input[name=tag-new]')
   }
-  get buttonAdd() {
+  get addButton() {
     return this.context.querySelector('button.add')
   }
   get suggestions() {
+    return this._suggestionsContainer.querySelectorAll('li')
+  }
+  get _languages() {
+    return document.querySelector('#language')
+  }
+  get _language() {
+    return this._languages.options[this._languages.selectedIndex].value
+  }
+  get _tags() {
+    return this._list.querySelectorAll('input')
+  }
+  get _tagsNames() {
+    return Array.from(this._tags).map(({ value }) => value)
+  }
+  get _list() {
+    return this.context.querySelector('.tags-list')
+  }
+  get _suggestionsContainer() {
     return this.context.querySelector('#tags-suggestions')
   }
 
   addNewTag(submission) {
-    if (!submission || tags._isTag(submission)) return
-    tags._addValue(submission)
-    tags._emptyAddField()
-    tags._inactiveSuggestionsList()
-    tags._render()
-  }
-  removeTag(tagElement) {
-    const submission = tagElement.innerText.trim()
-    if (!tags._isTag(submission)) return
-    tags._removeValue(submission)
-    tags._render()
+    if (!submission || this._isTagName(submission)) return
+    this._emptyAddField()
+    this._inactiveSuggestionsList()
+    return this._render(this._tagsNamesAdd(submission))
+  }
+  removeTag(submission) {
+    if (!this._isTagName(submission)) return
+    return this._render(this._tagsNamesRemove(submission))
   }
   handleSuggestion(submission) {
     this._request(submission).then(tagsApi =>
       this._createSuggestionList(tagsApi)
     )
   }
   _request(submission) {
-    return request(
-      `/article/tag/suggest/?language=${this.language}&terms=${submission}`
-    ).catch(console.error.bind(console))
+    const url = `/article/tag/suggest/?language=${
+      this._language
+    }&terms=${submission}`
+    return fetch(url)
+      .then(handleResponse)
+      .catch(error => {
+        const e = new Error(`${error.message} ${url}`)
```

Comment:
You should not have to do that as it is [already handled](https://github.com/cafebabel/cafebabel.com/blob/master/cafebabel/static/js/utils.js#L8) by the picolib we use, `error.url` should already be there 😉